src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderMatchAPIIntegrationTest.kt

```diff
@@ -221,6 +219,199 @@ internal class OffenderMatchControllerAPIIntegrationTest : AbstractTestExecution
+          .post("/match")
+          .then()
+          .statusCode(200)
+          .body("matches.findall.size()", equalTo(1))
+          .body("matches[0].offender.otherIds.crn", equalTo("X00001"))
+    }
+    @Test
+    internal fun `should match using PNC number if PNC and date of birth match`() {
+      given()
+          .auth()
+          .oauth2(jwtAuthenticationHelper.createJwt("ROLE_COMMUNITY"))
+          .contentType(MediaType.APPLICATION_JSON_VALUE)
+          .body(MatchRequest(
+              surname = "ahankara",
+              firstName = "adi",
+              dateOfBirth = LocalDate.of(1988, 1, 6),
+              pncNumber = "2015/0123456X",
+              activeSentence = true
+          ))
+          .post("/match")
+          .then()
+          .statusCode(200)
+          .body("matches.findall.size()", equalTo(1))
+          .body("matches[0].offender.otherIds.crn", equalTo("X00001"))
+    }
+
+  }
+
+  @Nested
+  inner class CRONumberMatching {
+    @BeforeEach
+    internal fun setup() {
+      loadOffenders(
+          OffenderIdentification(
+              surname = "gramsci",
+              firstName = "jan",
+              dateOfBirth = LocalDate.of(1988, 1, 6),
+              crn = "X00001",
+              croNumber = "SF80/655108T"
+          ),
+          OffenderIdentification(
+              surname = "gramsci",
+              firstName = "jan",
+              dateOfBirth = LocalDate.of(1988, 1, 6),
+              crn = "X888888",
+              croNumber = "SF80/655108T",
+              deleted = true
+          ),
+          OffenderIdentification(
+              surname = "gramsci",
+              firstName = "june",
+              dateOfBirth = LocalDate.of(1988, 1, 6),
+              crn = "X999999",
+              croNumber = "SF80/655108T",
+              activeSentence = false
+          ),
+          OffenderIdentification(
+              surname = "gramsci",
+              firstName = "jill",
+              dateOfBirth = LocalDate.of(1988, 1, 6),
+              crn = "X00002",
+              croNumber = "SF79/666108T",
+              nomsNumber = "G5555TT"
+          )
+      )
+    }
+    @Test
+    internal fun `should match using CRO number ignoring deleted and inactive sentences`() {
+      given()
+          .auth()
+          .oauth2(jwtAuthenticationHelper.createJwt("ROLE_COMMUNITY"))
+          .contentType(MediaType.APPLICATION_JSON_VALUE)
+          .body(MatchRequest(
+              surname = "gramsci",
+              firstName = "june",
+              dateOfBirth = LocalDate.of(1988, 1, 6),
+              croNumber = "SF80/655108T",
+              activeSentence = true
+          ))
+          .post("/match")
+          .then()
+          .statusCode(200)
+          .body("matches.findall.size()", equalTo(1))
+          .body("matches[0].offender.otherIds.crn", equalTo("X00001"))
+    }
+
+    @Test
+    internal fun `noms number takes precedence over CRO Number when present`() {
```

Comment:
How do you know that the nomsNumber took precedence?

Reply:
Very good question; so if we have a NOMS number it means a human has already matched this record. NOMS number only get into Delius by manually matching so this is really saying  "this record has already been linked" . Having said that I have definitely seen the occasional record that has been linked incorrectly - but for now decided not to tackle that particular edge case.